pa_matchbatch.sh

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env bash
+set -e
+
+# get rev hashes in the form [component]|[to_rev]|[url]
+getComponentRevMap() {
+	for componentName in $(for componentFile in $(grep -lv refs/tags configs/components/*.json); do grep -l puppetlabs ${componentFile}; done); do 
+		echo -n " $(basename ${componentName} .json)|"
+		ruby -rjson -e'j = JSON.parse(STDIN.read); print j["ref"] + "|" + j["url"]' < ${componentName}
+	done
+}
+
+getJiraProjectIdFor() {
+	case "${1}" in
+		facter) echo FACT
+		;;
+		hiera) echo HI
+		;;
+		leatherman) echo LTH
+		;;
+		puppet) echo PUP
+		;;
+		pxp-agent) echo PCP
+		;;
+		puppet-agent) echo PA
+		;;
+		*) (>&2 echo "Error: need to add JIRA project mapping for '${1}'.")
+			exit 1
+		;;
+	esac
+}
+
+getJiraFixedInFor() {
+	case "${1}" in
+		facter) echo FACT
+		;;
+		hiera) echo HI
+		;;
+		leatherman) echo LTH
+		;;
+		puppet) echo PUP
+		;;
+		pxp-agent) echo pxp-agent
+		;;
+		puppet-agent) echo puppet-agent
+		;;
+		*) (>&2 echo "Error: need to add JIRA fixed-in version mapping for '${1}'.")
+			exit 1
+		;;
+	esac
+}
+
+# main
+puppetAgentBaseRev=${1}
+[ -z "${puppetAgentBaseRev}" ] && { echo "Error: must specify the puppet-agent revision to start from."; exit 1; }
+
+cd ${WORKSPACE:-${PWD}}
+[ $(basename ${PWD}) == 'puppet-agent' ] || { echo "Error: must be in a puppet-agent directory"; exit 1; }
```

Comment:
I think you should also clone the agent if it does not exist. You can use git@github.com:puppetlabs/puppet-agent.git as the URL. Can add this to the repoRevMap.

Reply:
I like the idea of getting puppet-agent if you don't have it already, but
you need it to generate the initial repoRevMap, so it can't be inside the
loop. Chicken and egg problem. Switching to the desired tag can be, but not
the initial clone.